### ISSUE TITLE
[Order states] Controller code isn't following basic REST principles

### ISSUE THREAD

**Issue by filipefurtad0** [openfoodfoundation/openfoodnetwork#11122]
### What we should change and why (this is tech debt)

Comment: Opening as a tech-debt as, for users, the order is editable and can be found on the orders page.

Follow-up from the discussions occurring below (see Context).

The state of a backoffice order changes upon edit actions, which makes the order state move from `cart` to `payment`.

### Context

https://github.com/openfoodfoundation/openfoodnetwork/pull/10936#issuecomment-1580674643

https://github.com/openfoodfoundation/openfoodnetwork/issues/10719#issuecomment-1576551124

### Impact and timeline

Order state/controller consistency.

**Comment by filipefurtad0** [comment]
Yes, I think the original bug is fixed - so we should be good on that end.

But following a discussion with @abdellani, we've came to the conclusion that in backoffice orders, the order advances from `cart` to `payment` when the user views the order (clicks edit), if it has:
- line items
- addresses / customer id

I'm not sure it is correct that the view action triggers this change. I cannot reproduce this issue in staging, hence not a but and perhaps a tech debt.

@abdellani, can you confirm this summarizes our discussion correctly?

**Comment by abdellani** [comment]
There's one thing that I don't fully understand on `Spree::Admin::OrdersController#edit`
Why do we update the refresh rates on an action that's supposed to only load the page?

```ruby
      def edit
        @order.shipments.map(&:refresh_rates)
      end
```
